### PR TITLE
Added Fibonacci in Nim

### LINKED ISSUE #1669: Add Fibonacci in Nim

### PR DESCRIPTION
**Pull request by MichaelAOlson**
- [x] I fixed #1669

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### CODE REVIEW

File: archive/n/nim/fibonacci.nim [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Fibonacci Sample Program in Nim
+import strutils
+
+echo "Input the number of iterations (n): "
```
Comment (jrg94) [inline archive/n/nim/fibonacci.nim]: For this program, I'd take a peek at the requirements: https://sample-programs.therenegadecoder.com/projects/fibonacci/

We typically don't prompt for input. Instead, we pull it directly off the command line. Does Nim support command line args?

Comment (MichaelAOlson) [reply archive/n/nim/fibonacci.nim]: Yeah once I saw the build failed I started looking into that. There are a number of ways, just trying to figure out an os-agnostic method.

**Comment by jrg94** [comment]
Just took a peek at your tests. Everything looks great! At this point, it's an output formatting issue. We print all terms in the sequence in order by index:

```
1: 1
2: 1
3: 2
4: 3
5: 5
```

File: archive/n/nim/fibonacci.nim [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import strutils
+
+var n: BiggestUInt
+try:
+    n = paramStr(1).parseBiggestUInt
+except IndexError:
+    echo "Usage: please input the count of fibonacci numbers to output"
+    quit(1)
+except ValueError:
+    echo "invalid input: not a number"
```
Comment (jrg94) [inline archive/n/nim/fibonacci.nim]: We use the same string for both cases. Sorry if that was confusing in the table.

```suggestion
    echo "Usage: please input the count of fibonacci numbers to output"
```

**Comment by jrg94** [comment]
Was hello-world.nim intentionally deleted? Looks like it's going to get deleted when we merge this pull request.

File: archive/n/nim/fibonacci.nim (hunk 2 above)
Comment (MichaelAOlson1) [reply archive/n/nim/fibonacci.nim]: Updated

**Comment by MichaelAOlson** [comment]
Yes - this was the invalid file name that should have been tracked for
removal when it was renamed to hello_world.nim

**Pull request merged**